internal/lsp/server.go

```diff
@@ -233,6 +233,9 @@ func (s *Server) UnwatchFiles(ctx context.Context, handle project.WatcherHandle)
 
 // RefreshDiagnostics implements project.Client.
 func (s *Server) RefreshDiagnostics(ctx context.Context) error {
+	if s.initializeParams == nil || s.initializeParams.Capabilities == nil || s.initializeParams.Capabilities.Workspace == nil || s.initializeParams.Capabilities.Workspace.Diagnostics == nil {
```

Comment:
I would break this check into multiple lines and then also put the !ptrIsTrue here, making this func have just the one early return.

Reply:
I think what you have in the PR now is what I meant.